operator/README.md

```diff
@@ -0,0 +1,145 @@
+
+## Deploy karmada-operator
+
+Create a secret of kubeconfig to connect cluster.
+
+```shell
+kubectl create namespace karmada-system
+kubectl create secret generic my-kubeconfig --from-file=$HOME/.kube/config -n karmada-system
+```
+
+Deploy the karmada-operator workload.
+
+```shell
+kubectl apply -f operator/config/deploy/karmada-operator.yaml
+```
+
+The pod of karmada-operator in the `karmada-system` namespace will be running.
+
+```shell
+kubectl get po -n karmada-system
+NAME                               READY   STATUS    RESTARTS   AGE
+karmada-operator-5b7f485c5-g5lj5   1/1     Running   0          26s
+```
+
+## Create a Karmada instance
+
+A Karmada CR represents a Karmada instance, which is a namespace-scoped resource.
+The example below is to create a simple Karmada CR in the `test` namespace:
+
+```shell
+kubectl create namespace test
+kubectl apply -f - <<EOF
+apiVersion: operator.karmada.io/v1alpha1
+kind: Karmada
+metadata:
+  name: karmada-demo
+  namespace: test
+EOF
+```
+
+Wait for around 40 seconds, and the pods of the Karmada components will be running in the same namespace as the Karmada CR.
+
+```shell
+kubectl get po -n test
+karmada-demo-aggregated-apiserver-587bc5c697-v27vb      1/1     Running   0          12s
+karmada-demo-apiserver-55968d9f8c-mp8hf                 1/1     Running   0          35s
+karmada-demo-controller-manager-64455f7fd4-stls6        1/1     Running   0          5s
+karmada-demo-etcd-0                                     1/1     Running   0          37s
+karmada-demo-kube-controller-manager-584f978bbd-fftwq   1/1     Running   0          5s
+karmada-demo-scheduler-6d77b7547-hgz8n                  1/1     Running   0          5s
+karmada-demo-webhook-6f5944f5d8-bpkqz                   1/1     Running   0          5s
+```
+
+## Upgrade a Karmada instance
+
+Once a Karmada instance is created, the CR resource is automatically filled with default values.
+To upgrade the Karmada instance, for example, you can upgrade the Karmada version to v1.5.0 or newer:
+
+```shell
+kubectl patch karmada karmada-demo -n test --type merge -p '
+{
+  "spec": {
+    "components": {
+      "karmadaAggregatedAPIServer": {
+        "imageTag": "v1.5.0"
+      },
+      "karmadaControllerManager": {
+        "imageTag": "v1.5.0"
+      },
+      "karmadaScheduler": {
+        "imageTag": "v1.5.0"
+      },
+      "karmadaWebhook": {
+        "imageTag": "v1.5.0"
+      }
+    }
+  }
+}'
+```
+
+## Delete a Karmada instance
+
+A Karmada instance can be deleted once the Karmada CR is deleted.
+
+```shell
+kubectl delete karmada karmada-demo -n test
+```
+
+## Custom Karmada CR
+
+This feature allows you to configure the Karmada CR to install Karmada instances flexibly. For details, see [karmada.yaml](./config/samples/karmada.yaml).
```

Comment:
Could we add some more use cases?
eg. install karmada-descheduler/custom labels.